vms/platformvm/utxo/camino_locked_test.go

```diff
@@ -657,3 +662,787 @@ func TestVerifyLockUTXOs(t *testing.T) {
 		})
 	}
 }
+
+func TestGetDepositUnlockableAmounts(t *testing.T) {
+	config := defaultConfig()
+	ctx := snow.DefaultContextTest()
+	baseDBManager := db_manager.NewMemDB(version.Semantic1_0_0)
+	baseDB := versiondb.New(baseDBManager.Current().Database)
+	rewardsCalc := reward.NewCalculator(config.RewardConfig)
+	addr0 := ids.GenerateTestShortID()
+	addresses := ids.ShortSet{}
+	addresses.Add(addr0)
+
+	depositTxSet := ids.Set{}
+	testID := ids.GenerateTestID()
+	depositTxSet.Add(testID)
+
+	defaultState(config, ctx, baseDB, rewardsCalc)
+	tx := &dummyUnsignedTx{txs.BaseTx{}}
+	tx.Initialize([]byte{0})
+	outputOwners, _ := generateOwnersAndSig(tx)
+
+	depositedAmount := uint64(1000)
+	type args struct {
+		state        func(*gomock.Controller) state.Chain
+		depositTxIDs ids.Set
+		currentTime  uint64
+		addresses    ids.ShortSet
+	}
+	tests := map[string]struct {
+		args args
+		want map[ids.ID]uint64
+		err  error
+	}{
+		"Success retrieval of all unlockable amounts": {
+			args: args{
+				state: func(ctrl *gomock.Controller) state.Chain {
+					nowMinus20m := uint64(time.Now().Add(-20 * time.Minute).Unix())
+					s := state.NewMockChain(ctrl)
+					deposit1 := deposit.Deposit{
+						DepositOfferID:      testID,
+						UnlockedAmount:      0,
+						ClaimedRewardAmount: 5000,
+						Start:               nowMinus20m,
+						Duration:            uint32((10 * time.Minute).Seconds()),
+						Amount:              depositedAmount,
+					}
+					s.EXPECT().GetDeposit(testID).Return(&deposit1, nil)
+					s.EXPECT().GetDepositOffer(testID).Return(&deposit.Offer{
+						UnlockHalfPeriodDuration: uint32((10 * time.Minute).Seconds()),
+						Start:                    nowMinus20m,
+					}, nil)
+					return s
+				},
+				depositTxIDs: depositTxSet,
+				currentTime:  uint64(time.Now().Unix()),
+				addresses:    outputOwners.AddressesSet(),
+			},
+			want: map[ids.ID]uint64{testID: depositedAmount},
+		},
+		"Success retrieval of 50% unlockable amounts": {
+			args: args{
+				state: func(ctrl *gomock.Controller) state.Chain {
+					nowMinus20m := uint64(time.Now().Add(-20 * time.Minute).Unix())
+					s := state.NewMockChain(ctrl)
+					deposit1 := deposit.Deposit{
+						DepositOfferID:      testID,
+						UnlockedAmount:      0,
+						ClaimedRewardAmount: 5000,
+						Start:               nowMinus20m,
+						Duration:            uint32((20 * time.Minute).Seconds()),
+						Amount:              depositedAmount,
+					}
+					s.EXPECT().GetDeposit(testID).Return(&deposit1, nil)
+					s.EXPECT().GetDepositOffer(testID).Return(&deposit.Offer{
+						UnlockHalfPeriodDuration: uint32((10 * time.Minute).Seconds()),
+						Start:                    nowMinus20m,
+					}, nil)
+					return s
+				},
+				depositTxIDs: depositTxSet,
+				currentTime:  uint64(time.Now().Unix()),
```

Comment:
I'd define some common var "currentTime" outside of test cases and use that across all of them
Also, make deposit/offer start-time use this var in order to not call time.Now() twice, so we'll be 100% sure it won't differ